lib/slack/web/faraday/request.rb

```diff
@@ -28,11 +28,12 @@ def request(method, path, options)
               request.url(path, options)
             when :post, :put
               request.path = path
-              options.compact!
+              options.compact! unless options.is_a? String
               request.body = options unless options.empty?
             end
             request.headers['Authorization'] = "Bearer #{token}" if token
-            request.options.merge!(options.delete(:request)) if options.key?(:request)
+
+            request.options.merge!(options.delete(:request)) if !options.is_a?(String) && options.key?(:request)
```

Comment:
I would prefer a positive test here too.

```suggestion
            request.options.merge!(options.delete(:request)) if !options.respond_to?(:key) && options.key?(:request)
```
(or possibly check if `options.is_a?(Hash)`.

Reply:
Agreed https://github.com/slack-ruby/slack-ruby-client/pull/532/commits/6e103a47f0a52adaa8cc567b730a3881df8a5022